### PR TITLE
Julian.mainardis/unit testing

### PR DESCRIPTION
**Pull request by julian-mainardis-globant**
## Description

Add all the unit testings.

## Story Type

- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?

Run all the unit tests one by one or all together to verify that all pass without errors.

## Evidences

Coverage:
image
image

## Checklist

- All unit Test passed :white_check_mark:

## Package Structure
image

### CODE REVIEW

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/ReservationActivity.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.mobile.tandil.javabaseproject.databinding.ActivityReservationBinding;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+import com.mobile.tandil.javabaseproject.mvp.model.ReservationModel;
+import com.mobile.tandil.javabaseproject.mvp.presenter.ReservationPresenter;
+import com.mobile.tandil.javabaseproject.mvp.view.ReservationView;
+import java.util.Calendar;
+
+public class ReservationActivity extends AppCompatActivity implements ListenerPickers {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/activity/ReservationActivity.java]: Please delete this empty space

File: app/src/main/java/com/mobile/tandil/javabaseproject/database/ParkingDatabase.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package com.mobile.tandil.javabaseproject.database;
+
+import com.mobile.tandil.javabaseproject.entity.Reservation;
+import com.mobile.tandil.javabaseproject.util.Constants;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class ParkingDatabase {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/database/ParkingDatabase.java]: Please delete this empty space

File: app/src/main/java/com/mobile/tandil/javabaseproject/entity/Reservation.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.mobile.tandil.javabaseproject.entity;
+
+import com.mobile.tandil.javabaseproject.util.DateUtils;
+import java.util.Calendar;
+
+public class Reservation {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/entity/Reservation.java]: Same here

File: app/src/main/java/com/mobile/tandil/javabaseproject/fragment/ReservationDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import android.view.ViewGroup;
+import androidx.annotation.Nullable;
+import com.mobile.tandil.javabaseproject.databinding.DialogFragmentPickersBinding;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.PickersContract;
+import com.mobile.tandil.javabaseproject.mvp.presenter.PickersPresenter;
+import com.mobile.tandil.javabaseproject.mvp.view.PickersView;
+
+public class ReservationDialogFragment extends DialogFragment {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/fragment/ReservationDialogFragment.java]: Same here

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenter.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.mobile.tandil.javabaseproject.mvp.presenter;
+
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.PickersContract;
+
+public class PickersPresenter implements PickersContract.Presenter {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenter.java]: Please remove this empty space

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
+package com.mobile.tandil.javabaseproject.mvp.presenter;
+
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+import com.mobile.tandil.javabaseproject.util.DateUtils;
+import com.mobile.tandil.javabaseproject.util.ValidationEnum;
+import java.util.Calendar;
+
+public class ReservationPresenter implements ReservationContract.Presenter {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java]: Same here

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ReservationView.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.annotation.StringRes;
+import com.mobile.tandil.javabaseproject.R;
+import com.mobile.tandil.javabaseproject.databinding.ActivityReservationBinding;
+import com.mobile.tandil.javabaseproject.fragment.ReservationDialogFragment;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+import com.mobile.tandil.javabaseproject.mvp.view.base.ActivityView;
+
+public class ReservationView extends ActivityView implements ReservationContract.View {
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ReservationView.java]: Same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/DialogPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DialogPresenterTest {
+
```
Comment (Georgina-Fernandez) [inline app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/DialogPresenterTest.java]: Same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ParkingPresenterTest {
+
```
Comment (Georgina-Fernandez) [inline app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenterTest.java]: Same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenterTest.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PickersPresenterTest {
+
```
Comment (Georgina-Fernandez) [inline app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenterTest.java]: Same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenterTest.java [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.util.Calendar;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReservationPresenterTest {
+
```
Comment (Georgina-Fernandez) [inline app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenterTest.java]: Same here

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/ReservationActivity.java (hunk 1 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/activity/ReservationActivity.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/database/ParkingDatabase.java (hunk 2 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/database/ParkingDatabase.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/entity/Reservation.java (hunk 3 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/entity/Reservation.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/fragment/ReservationDialogFragment.java (hunk 4 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/fragment/ReservationDialogFragment.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenter.java (hunk 5 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenter.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java (hunk 6 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ReservationView.java (hunk 7 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ReservationView.java]: done

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/DialogPresenterTest.java (hunk 8 above)
Comment (julian-mainardis-globant) [reply app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/DialogPresenterTest.java]: done

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenterTest.java (hunk 9 above)
Comment (julian-mainardis-globant) [reply app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenterTest.java]: done

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenterTest.java (hunk 10 above)
Comment (julian-mainardis-globant) [reply app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/PickersPresenterTest.java]: done

File: app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenterTest.java (hunk 11 above)
Comment (julian-mainardis-globant) [reply app/src/test/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenterTest.java]: done

**Review by Georgina-Fernandez — APPROVED** [review]

**Review by jnicolaslaurenti — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by kearneyeduardo — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**